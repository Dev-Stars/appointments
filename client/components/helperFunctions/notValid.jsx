const notValid = (event) => {
  if (event) {
    event.preventDefault();
  }
  alert('Must complete the required fields.');
};

export default notValid;
