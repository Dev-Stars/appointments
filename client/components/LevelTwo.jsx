import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import notValid from './helperFunctions/notValid.jsx';

const Title = styled.h1`
  font-size: 20px;
  font-weight: 600;
  color: #d90007;
`;

const FormCategory = styled.div`
  width: 100%;
  margin-top: 10px;
  font-size: 12.5px;
`;

const InputStyle = styled.input`
  width: 100%;
  font-size: 12.5px;
  padding: 6px 10px;
  margin: 8px 0;
  display: inline-block;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-sizing: border-box;
`;

const Summary = styled.div`
  margin-top: 20px;
  margin-bottom: 20px;
  width: 90%;
  padding: 5%;
  padding-top: 10%;
  background: #f5f5f5;
  border: 1px solid #e6e6e6;
  border-radius: 4px;
  font-size: 14px;
  font-family: Helvetica Neue,Helvetica,Arial,sans-serif;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: minmax(40px, auto);
`;

const SummaryGrid = styled.div`
  grid-column: ${props => props.column};
  grid-row: ${props => props.row};
`;

const Edit = styled.div`
  grid-column: 1 / 5;
  grid-row: 4;
  padding-top: 15px;
  color: #0073bb;
  width: 0;
`;

const Button = styled.button`
  width: 100%;
  color: white;
  background: #d90007;
  border-color: #8d0005;
  box-shadow: 0 1px 1px rgba(0,0,0,0.3);
  border-radius: 5px;
  padding: 5px 8px;
  font-size: 12px;
  font-weight: 600;
  line-height: 1.5em;
  height: 34px;
`;

const LevelTwo = (props) => {
  const {
    name, address, city, state, zip, change, date, time, book,
    first, last, email, phone, count, back,
  } = props;

  const addressString = `${city} ${state}, ${zip}`;

  let isValid = false;
  if (first && last && email && phone) {
    isValid = true;
  }

  return (
    <form>
      <div>
        <Title>
          Confirm Reservation
        </Title>
        <FormCategory>
          First Name*
          <InputStyle id="inputFirstName" onChange={change} name="firstName" value={first} />
        </FormCategory>
        <FormCategory>
          Last Name*
          <InputStyle id="inputLastName" onChange={change} name="lastName" value={last} />
        </FormCategory>
        <FormCategory>
          Email*
          <InputStyle id="inputEmail" onChange={change} name="email" value={email} />
        </FormCategory>
        <FormCategory>
          Mobile Number*
          <InputStyle id="inputPhone" onChange={change} name="phone" value={phone} />
        </FormCategory>
        <FormCategory>
          *required field
        </FormCategory>
      </div>
      <Summary>
        <SummaryGrid id="whenSummaryKey" column={'1 / 5'} row={'1'}>
          When:
        </SummaryGrid>
        <SummaryGrid id="whenSummaryPropert" column={'2 / 5'} row={'1'}>
          {date}
          {' at '}
          {time}
        </SummaryGrid>
        <SummaryGrid id="guestCountSummaryKey" column={'1 / 5'} row={'2'}>
          Party:
        </SummaryGrid>
        <SummaryGrid id="guestCountSummaryProperty" column={'2'} row={'2'}>
          <div id="countCorrect">
            {count}
            {' '}
            {count === '1' ? 'Person' : 'People'}
          </div>
        </SummaryGrid>
        <SummaryGrid id="placeSummaryKey" column={'1 / 5'} row={'3'}>
          Place:
        </SummaryGrid>
        <SummaryGrid id="placeSummaryProperty" column={'2 / 5'} row={'3 / 5'}>
          {name}
          <br />
          {address}
          <br />
          {addressString}
        </SummaryGrid>
        <Edit id="editSummary" role="button" tabIndex={0} onClick={back} onKeyPress={back}>
          EDIT
        </Edit>
      </Summary>
      {isValid
        ? (
          <Button id="submitValid" type="submit" onClick={book}>
            Book Appointment
          </Button>
        )
        : (
          <Button id="submitInvalid" type="submit" onClick={notValid}>
            Not complete
          </Button>
        )
      }
    </form>
  );
};

LevelTwo.propTypes = {
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  state: PropTypes.string.isRequired,
  zip: PropTypes.string.isRequired,
  change: PropTypes.func.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  book: PropTypes.func.isRequired,
  first: PropTypes.string.isRequired,
  last: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  phone: PropTypes.string.isRequired,
  count: PropTypes.string.isRequired,
  back: PropTypes.func.isRequired,
};

export default LevelTwo;
