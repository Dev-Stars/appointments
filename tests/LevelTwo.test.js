import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import * as enzyme from 'enzyme';
import React from 'react';
import LevelTwo from '../client/components/LevelTwo.jsx';
import notValid from '../client/components/helperFunctions/notValid.jsx'

enzyme.configure({ adapter: new Adapter() });

describe('LevelTwo, a child component to Appointment', () => {

  it('should import a notValid function', () => {
    const type = typeof notValid;
    expect(type).toEqual('function');
    notValid();
    expect(notValid()).toEqual(undefined);
  });

  it('should call back function when edit is clicked', () => {
    const spyFunction = jest.fn();
    let wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={(() => { null })()}
        book={(() => { null })()}
        first={"firstName"}
        last={"lastName"}
        email={"email"}
        phone={"phone"}
        count={"1"}
        back={spyFunction}
      />
    );
    wrapper.find('div#edit').simulate('click');
    expect(spyFunction).toHaveBeenCalled();
  });

  it('handle change should be called when each input field has text added', () => {
    const spyFunction = jest.fn();
    let wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={spyFunction}
        book={(() => { null })()}
        first={"firstName"}
        last={"lastName"}
        email={"email"}
        phone={"phone"}
        count={"1"}
        back={(() => { null })()}
      />
    );
    wrapper.find('input#inputFirstName').simulate('change');
    wrapper.find('input#inputLastName').simulate('change');
    wrapper.find('input#inputEmail').simulate('change');
    wrapper.find('input#inputPhone').simulate('change');

    expect(spyFunction).toHaveBeenCalledTimes(4);
  });



  it('should activate the submit function if forms are complete', () => {
    const spyFunction = jest.fn();
    let wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={(() => { null })()}
        book={spyFunction}
        first={"firstName"}
        last={"lastName"}
        email={"email"}
        phone={"phone"}
        count={"1"}
        back={(() => { null })()}
      />
    );
    wrapper.find('button').simulate('click');
    expect(spyFunction).toHaveBeenCalled();
  })

  it('should render not complete function if fields are not complete', () => {
    const spyFunction = jest.fn();
    let wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={(() => { null })()}
        book={spyFunction}
        first={""}
        last={""}
        email={""}
        phone={""}
        count={"1"}
        back={(() => { null })()}
      />
    );
    wrapper.find('button').simulate('click');
    expect(spyFunction).not.toHaveBeenCalled();
  });

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={(() => { null })()}
        book={(() => { null })()}
        first={"firstName"}
        last={"lastName"}
        email={"email"}
        phone={"phone"}
        count={"1"}
        back={(() => { null })()}
      />
    );
  });

  it('should render one form enclosing all elements', () => {
    expect(wrapper.find('form').exists()).toEqual(true);
  });

  it('should render 4 input fields an edit and a submit button', () => {
    expect(wrapper.find('input#inputFirstName').exists()).toEqual(true);
    expect(wrapper.find('input#inputLastName').exists()).toEqual(true);
    expect(wrapper.find('input#inputEmail').exists()).toEqual(true);
    expect(wrapper.find('input#inputPhone').exists()).toEqual(true);
    expect(wrapper.find('button').exists()).toEqual(true);
  });

  it('should say person, if reservation is for 1, people if otherwise', () => {
    expect(wrapper.find('div#guestCountSummary').text()).toEqual('1 Person');

    const numberOfPeople = '8';
    wrapper = shallow(
      <LevelTwo
        date={"July 26, 2018"}
        time={"10:30 AM"}
        name={"Cakesons"}
        address={"123"}
        city={"businessCity"}
        state={"businessState"}
        zip={"businessZip"}
        change={(() => { console.log('dummy') })()}
        book={(() => { console.log('dummy') })()}
        first={"firstName"}
        last={"lastName"}
        email={"email"}
        phone={"phone"}
        count={numberOfPeople}
        back={(() => { console.log('dummy') })()}
      />
    );
    expect(wrapper.find('div#guestCountSummary').text()).toEqual(numberOfPeople+ ' People');
  });
});
