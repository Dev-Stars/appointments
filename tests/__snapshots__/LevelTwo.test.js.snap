// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LevelTwo, a child component to Appointment should render 4 input fields with correct titles 1`] = `
<form>
  <div>
    <div
      style={
        Object {
          "color": "#d90007",
          "fontSize": "20px",
          "fontWeight": "600",
        }
      }
    >
      Confirm Reservation
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "12px",
            "marginTop": "10px",
            "width": "30%",
          }
        }
      >
        First Name*
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          id="inputFirstName"
          name="firstName"
          style={
            Object {
              "width": "100%",
            }
          }
          value="firstName"
        />
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "12px",
            "marginTop": "10px",
            "width": "30%",
          }
        }
      >
        Last Name*
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          id="inputLastName"
          name="lastName"
          style={
            Object {
              "width": "100%",
            }
          }
          value="lastName"
        />
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "12px",
            "marginTop": "10px",
            "width": "30%",
          }
        }
      >
        Email*
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          id="inputEmail"
          name="email"
          style={
            Object {
              "width": "100%",
            }
          }
          value="email"
        />
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "12px",
            "marginTop": "10px",
            "width": "30%",
          }
        }
      >
        Mobile Number*
      </div>
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <input
          id="inputPhone"
          name="phone"
          style={
            Object {
              "width": "100%",
            }
          }
          value="phone"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "fontSize": "12px",
          "marginTop": "10px",
          "width": "30%",
        }
      }
    >
      *required field
    </div>
  </div>
  <div
    style={
      Object {
        "background": "lightGrey",
        "borderColor": "grey",
        "borderRadius": "5px",
        "display": "grid",
        "fontSize": "12px",
        "gridAutoRows": "minmax(40px, auto)",
        "gridTemplateColumns": "repeat(4, 1fr)",
        "marginBottom": "20px",
        "marginTop": "20px",
        "padding": "5%",
        "paddingTop": "10%",
        "width": "90%",
      }
    }
  >
    <div
      syle={
        Object {
          "gridColumn": "1 / 5",
          "gridRow": "1",
        }
      }
    >
      When:
    </div>
    <div
      style={
        Object {
          "gridColumn": "2 / 5",
          "gridRow": "1",
        }
      }
    >
      July 26, 2018
       at 
      10:30 AM
    </div>
    <div
      style={
        Object {
          "gridColumn": "1 / 5",
          "gridRow": "3",
        }
      }
    >
      Place:
    </div>
    <div
      style={
        Object {
          "gridColumn": "2 / 5",
          "gridRow": "3/5",
        }
      }
    >
      Cakesons
      <br />
      123
      <br />
      businessCity businessState, businessZip
    </div>
    <div
      syle={
        Object {
          "gridColumn": "1 / 5",
          "gridRow": "2",
        }
      }
    >
      Party:
    </div>
    <div
      id="guestCountSummary"
      style={
        Object {
          "gridColumn": "2",
          "gridRow": "2",
        }
      }
    >
      1
       
      Person
    </div>
    <div
      style={
        Object {
          "color": "#0073bb",
          "gridColumn": "1 / 5",
          "gridRow": "4",
          "paddingTop": "15px",
        }
      }
    >
      <div
        id="edit"
        role="button"
        tabIndex={0}
      >
        EDIT
      </div>
    </div>
  </div>
  <button
    style={
      Object {
        "background": "#d90007",
        "borderColor": "#8d0005",
        "borderRadius": "5px",
        "boxShadow": "0 1px 1px rgba(0,0,0,0.3)",
        "color": "white",
        "fontSize": "12px",
        "fontWeight": "600",
        "height": "34px",
        "lineHeight": "1.5em",
        "padding": "5px 8px",
        "width": "100%",
      }
    }
    type="submit"
  >
    Book Appointment
  </button>
</form>
`;
